src/api-binder.coffee

```diff
@@ -83,6 +86,21 @@ module.exports = class APIBinder
 				passthrough: passthrough
 			@cachedBalenaApi = @balenaApi.clone({}, cache: {})
 
+	loadBackupFromMigration: (retryDelay) =>
+		pathExistsOnHost(path.join('mnt/data', constants.migrationBackupFile))
+		.then (exists) =>
+			if !exists
+				return
+			console.log('Migration backup detected')
+			@getTargetState()
+			.then (targetState) =>
+				@deviceState.restoreBackup(targetState)
+		.catch (err) =>
+			console.log('Error restoring migration backup, retrying: ', err)
+			Promise.delay(retryDelay)
+			.then =>
+				@loadBackupFromMigration(retryDelay)
```

Comment:
Won't this constantly loop if the backup cannot be loaded?

Reply:
@CameronDiver yes... we don't want to continue if we can't load the backup (if there is a backup, that is). I guess we can instead throw the exception and let the supervisor healthcheck kick in? Though probably the effect will be the same, and with this loop we can probably catch transient errors, e.g. a network error when getting the target state